tuf/utils/x509.go

```diff
@@ -428,18 +428,18 @@ func blockType(k data.PrivateKey) (string, error) {
 }
 
 // KeyToPEM returns a PEM encoded key from a Private Key
-func KeyToPEM(privKey data.PrivateKey, role, gun string) ([]byte, error) {
+func KeyToPEM(privKey data.PrivateKey, role data.RoleName, gun data.GUN) ([]byte, error) {
 	bt, err := blockType(privKey)
 	if err != nil {
 		return nil, err
 	}
 
 	headers := map[string]string{}
-	if role != "" {
-		headers["role"] = role
+	if role.String() != "" {
```

Comment:
No need to call `.String()`, Go's compiler will work out that you mean the literal `""` to be of type `data.RoleName`. Same applies to `if gun.String() != ""` a couple of lines down.